app/controllers/case_workers/admin/management_information_controller.rb

```diff
@@ -30,13 +31,32 @@ def generate
     redirect_to case_workers_admin_management_information_url, flash: { notification: message }
   end
 
+  def create
+    StatsReportGenerationJob.perform_later(report_params[:report_type], day: @day)
+    message = t('case_workers.admin.management_information.job_scheduled')
+    redirect_to case_workers_admin_management_information_url, alert: message
+  end
+
   private
 
   def validate_report_type
-    return if Stats::StatsReport::TYPES.include?(params[:report_type])
+    return if Stats::StatsReport.names.include?(params[:report_type])
     redirect_to case_workers_admin_management_information_url, alert: t('.invalid_report_type')
   end
 
+  def report_params
+    params.permit(
+      :report_type,
+      :day
+    )
+  end
+
+  def validate_and_set_date
+    @day ||= Date.iso8601("#{report_params['day(1i)']}-#{report_params['day(2i)']}-#{report_params['day(3i)']}")
```

Comment:
There are options and here and Date.new is one (also looked Date.civil). I ended up with this as this is the format that that needs sending to the SQL query. In the end as long as it Validates the date and raises an error when invalid then it should be fine.


the above method  requires leading zeros, so not ideal. `Date.new` as above would allow negative numbers, which is not ideal either.

Will have a play and test

Reply:
> I ended up with this as this is the format that that needs sending to the SQL query.

Unless I am missing something,

```
Date.iso8601("#{report_params['day(1i)']}-#{report_params['day(2i)']}-#{report_params['day(3i)']}")
Date.new(report_params['day(1i)].to_i, report_params['day(2i)'].to_i, report_params['day(3i)'].to_i)
```

are identical. Both can be made into an ISO8601 with `@day.iso8601`.

> In the end as long as it Validates the date and raises an error when invalid then it should be fine.

That's true. I was just wondering why it was being formatted so that it could then be parsed to create the Date instance.

Reply:
I have changed to use Date.new to allow leading or non-leading zeros, as that might be  a pain for users

the slice works as below, which feels to messy

```
Date.new(*report_params.to_h.slice('start_at(1i)', 'start_at(2i)', 'start_at(3i)').values.map(&:to_i))
```

Reply:
see fc677ca19